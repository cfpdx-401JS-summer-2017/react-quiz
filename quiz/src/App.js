import React, { Component } from 'react';
import './App.css';
import { Number } from './Number';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      number: 1
    }
    this.increaseNumber.bind(this);
    this.decreaseNumber.bind(this);
  }

  increaseNumber(number) {
    this.setState({
      number: this.state.number + 1
    });
  }
  decreaseNumber(number) {
    this.setState({
      number: this.state.number - 1
    });
  }


  render() {
    return (
      <div>
        <button disabled={this.state.number = 0} onClick={() => this.setState({number: this.state.number - 1})}>-</button>
        <Number number={this.state.number} />
        <button onClick={() => this.setState({number: this.state.number + 1})}>+</button>
      </div>
    );
  }
}

export default App;
