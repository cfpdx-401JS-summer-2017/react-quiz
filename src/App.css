.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

#main {
  border: 1px solid #999;
  border-radius: 10px;
  padding: 15px 30px;
  display: inline-block;
  margin-top: 10px;
}
#main div {display: inline-block; margin: 0 10px;}
#main div, #main button {font-size: 1.5em;}
#main button {
  background: #fff;
  border: 1px solid #999;
}
#main button.disabled {
  background: #999;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
