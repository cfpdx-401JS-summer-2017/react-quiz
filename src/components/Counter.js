import React from 'react';
import PropTypes from 'prop-types';


export default function Counter(props) {
  Counter.propTypes = {
    currentVal: PropTypes.number
  }

  return (
    <div>
      {props.currentVal > 0 &&
      <div onClick={()=> props.updateCounter('dec')}>minus one</div>}
      <div>
        {props.currentVal}
      </div>
      <div onClick={() => props.updateCounter('inc')}>plus one</div>
    </div>
  );
}
