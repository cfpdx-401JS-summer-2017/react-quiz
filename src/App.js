import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const Num = 0;
class App extends Component {
  constructor() {
    super();
    this.state = {
      increment: 0
    }
  }
  render() {
    const { increment } = this.state;
    function Incrementor(increment) {
      return increment;

    }
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
            <section>
            <Incrementor increment={increment}/>
            </section>
          <section>
            <button 
              onClick={() => this.setState({increment:this.state.increment-1})}
              style={{padding:'10px',fontSize:20}}
              disabled={this.state.increment === 0}
              >-
            </button>
            <button 
              onClick={() =>  this.setState({{increment:this.state.increment+1})}
              style={{padding: '10px',fontSize:20}}
            >+
            </button>
          </section>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
