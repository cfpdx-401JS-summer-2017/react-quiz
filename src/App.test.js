import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Counter from '../src/components/Counter';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot when the value is > 0', () => {
  const tree = renderer.create(<Counter currentVal={2} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a snapshot when the value === 0', () => {
  const tree = renderer.create(<Counter currentVal={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});