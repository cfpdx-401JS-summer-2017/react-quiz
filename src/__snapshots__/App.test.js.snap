// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot when the value === 0 1`] = `
<div>
  <div>
    0
  </div>
  <div
    onClick={[Function]}
  >
    plus one
  </div>
</div>
`;

exports[`renders a snapshot when the value is > 0 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    minus one
  </div>
  <div>
    2
  </div>
  <div
    onClick={[Function]}
  >
    plus one
  </div>
</div>
`;
